### PR TITLE
fix: reuse instances in configurationBeanLoader to prevent stack over…

### PR DESCRIPTION
**Pull request by litvinovg**
**[VIVO GitHub issue](https://github.com/vivo-project/VIVO/issues/3675)**: (please link to issue)

- Added map of created objects to prevent creation of duplicates for each requested instance.
- Adds loop test for configuration bean loader

# Interested parties
Tag (@ mention) interested parties or, if unsure, @VIVO-project/vivo-committers

### CODE REVIEW

File: api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoader.java [hunk 1]
```diff
@@ -135,22 +140,38 @@ private ConfigurationBeanLoader(LockableModel locking, ServletContext ctx,
...
 			throw new NullPointerException("uri may not be null.");
 		}
 		if (resultClass == null) {
 			throw new NullPointerException("resultClass may not be null.");
 		}
-
+		if (instancesMap.containsKey(uri)) {
+		  try {
+		    T t = (T) instancesMap.get(uri);
+        return t;
```
Comment (chenejac) [inline api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoader.java:161]: Can you please correct layouting of this line

File: api/src/test/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoaderTest.java [hunk 2]
```diff
@@ -394,6 +389,19 @@ public void simpleSuccessIgnoringExtraProperties() throws ConfigurationBeanLoade
 	public static class SimpleSuccess {
 		// Nothing of interest.
 	}
+	
+  public static class Friend {
```
Comment (chenejac) [inline api/src/test/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoaderTest.java:393]: Please correct indentation

**Review by chenejac — CHANGES_REQUESTED** [review]
@litvinovg I have just posted two comments related to indentation. Of course, it is not critical, but it would be nice if you can correct that. It is great that contribution is covered with test.

File: api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoader.java (hunk 1 above)
Comment (litvinovg) [reply api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoader.java]: Fixed. Thanks.

File: api/src/test/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoaderTest.java (hunk 2 above)
Comment (litvinovg) [reply api/src/test/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoaderTest.java]: Fixed. Thanks.

**Review by chenejac — CHANGES_REQUESTED** [review]
Tests are failing for this PR, can you please check before we merge?

**Comment by wwelling** [comment]
@chenejac I have pulled this down are ran tests. They pass consistently. Also rerunning action passed them.

**Review by wwelling — APPROVED** [review]

**Review by chenejac — APPROVED** [review]

**Pull request merged**